packages/framework-provider-aws/src/library/graphql-adapter.ts

```diff
@@ -1,26 +1,36 @@
 import { APIGatewayProxyEvent } from 'aws-lambda'
-import { GraphQLRequestEnvelope, Logger } from '@boostercloud/framework-types'
-import { fetchUserFromRequest } from './user-envelopes'
+import { GraphQLRequestEnvelope, GraphQLRequestEnvelopeError, Logger } from '@boostercloud/framework-types'
 import { CognitoIdentityServiceProvider } from 'aws-sdk'
+import { userEnvelopeFromAuthToken } from './auth-adapter'
 
 export async function rawGraphQLRequestToEnvelope(
   userPool: CognitoIdentityServiceProvider,
   request: APIGatewayProxyEvent,
   logger: Logger
-): Promise<GraphQLRequestEnvelope> {
+): Promise<GraphQLRequestEnvelope | GraphQLRequestEnvelopeError> {
```

Comment:
So what happened here:

This function is called by the core when a GraphQL request comes either through HTTP or WebSockets. 
When everything goes correctly, it returns the GraphQL request envelope with information to know the connection ID of the socket if it is the case. 
When something goes wrong like, for example, the token supplied by the user is expired and the function `userEnvelopeFromAuthToken` throws an error, we were just bubbling up the error. But that's wrong, as we also need to know the connectionID of the socket so that we can send the error back to the client, and that's what we are doing here now (we are also including more information)

The reason why returning a union type (Correct Result | Error result) it simplifies a lot the logic in the core. I tried to throw a custom error with the connectionID as a field, but the code in the core ended-up spaghettified (the reasons were that when you catch an error, you can't specify its type and that this error is happening before the core knows the structure of the request, complicating everything)

Reply:
This fixes several problems. Now, on the client side, you can properly know when the auth token is expired and refresh it, as you will get an expected error. Before, the client was getting a "network error" and the GraphQL client engine failed.